kubernetes-itests/src/test/java/io/fabric8/kubernetes/NetworkPolicyIT.java

```diff
@@ -158,16 +165,25 @@ public void update(){
 
   @Test
   public void delete(){
+    Ready<NetworkPolicy> networkPolicyReady = new Ready<NetworkPolicy>(NetworkPolicy.class, client, "networkpolicy", currentNamespace);
+    await().atMost(30, TimeUnit.SECONDS).until(networkPolicyReady);
     boolean deleted = client.network().networkPolicies().delete(networkPolicy);
+
     assertTrue(deleted);
     NetworkPolicyList networkPolicyList = client.network().networkPolicies().list();
     assertEquals(0,networkPolicyList.getItems().size());
   }
 
   @After
   public void cleanup() throws InterruptedException {
-    client.network().networkPolicies().delete();
-    // wait for graceful deletion
-    Thread.sleep(6 * 1000);
+    try {
+      if (client.network().networkPolicies().list().getItems().size()!= 0) {
```

Comment:
`client.network().networkPolicies()` can be a variable to start with. Then perhaps `!var.list().isEmpty()` might improve readability a bit?

Reply:
Hey, 

Attempting this would look like [this](https://gist.github.com/rohanKanojia/f49dd6ce84b9317fb21f8653e92e8c81#file-networkpolicyit-java-L47) and i think this line looks really long and highly unreadable. I would be happy to take any suggestions from you on workarounds.